compiler/codegen/codegen.cpp

```diff
@@ -2508,6 +2515,116 @@ static void codegenGpuGlobals() {
 }
 #endif
 
+#ifdef HAVE_LLVM
+// Implements the RemarkSerializer interface for LLVM
+// For each remark to be outputted (after filtering for pass names), `emit()` is called
+// We can then do further filtering and  control how the remarks are printed
+struct ChapelRemarkSerializer : public llvm::remarks::RemarkSerializer {
+  ChapelRemarkSerializer(llvm::raw_ostream& OS)
+      : llvm::remarks::RemarkSerializer(
+            llvm::remarks::Format::Unknown, OS,
+            llvm::remarks::SerializerMode::Standalone) {}
+
+  void emit(const llvm::remarks::Remark& Remark) override {
+
+    llvm::StringRef funcCName = Remark.FunctionName;
+    const char* astr_funcCName = astr(funcCName.str());
+    auto funcIt = gGenInfo->functionCNameAstrToSymbol.find(astr_funcCName);
+    FnSymbol* fn = nullptr;
+    if (funcIt != gGenInfo->functionCNameAstrToSymbol.end()) {
+      fn = funcIt->second;
+    }
+
+    // couldn't get from map, do the slow way with a loop through the FnSymbols
```

Comment:
Why would it need to do this? Seems like the loop below, per-remark, could add a lot of compile time.

Reply:
Any case where the function name isn't added to `functionCNameAstrToSymbol`. This is currently skipped for anything where `getFunctionDeclClang(cname) != nullptr`.

Its not a common case, but there are a few runtime functions which fall into this category.

Reply:
Thinking about it, after all this function name searching if `fn` is still `nullptr` and not in developer mode the function gets skipped. So one option would be to only do this extra work if in developer mode, to avoid impacting most users.

I would also be ok with just removing it and accepting that some functions may not get as nice of location information when in developer mode.

Reply:
Of these, I'd lean towards removing it (for now) and replacing it with a TODO comment. For one thing, perhaps the thing to do with those runtime functions is to loop through the FnSymbols and update the map.